addons/stock/models/stock_move.py

```diff
@@ -1621,9 +1621,17 @@ def _action_assign(self, force_qty=False):
         moves_to_redirect = OrderedSet()
         moves_to_assign = self
         if not force_qty:
-            moves_to_assign = moves_to_assign.filtered(
-                lambda m: not m.picked and m.state in ['confirmed', 'waiting', 'partially_available']
-            )
+            if hasattr(self, 'raw_material_production_id') and \
+                    all(not (move.move_line_ids and move.quantity > 0) for move in self) and \
+                    any(code == 'mrp_operation' for code in self.raw_material_production_id.mapped('picking_type_id.code')) and \
+                    any(raw_material_production.qty_producing > 0 for raw_material_production in self.raw_material_production_id):
+                moves_to_assign = moves_to_assign.filtered(
+                    lambda m: m.state in ['confirmed', 'waiting', 'partially_available']
+                )
```

Comment:
This isn't right, you shouldn't add code from `mrp` directly in `stock`. This is bad practice and makes it hell to maintain.

What are you trying to achieve here? Why do you want to force-assign moves that are already picked?

Reply:
Hello @clesgow,

I added this condition because when we use the shop floor to add a component in a MO, the previous condition prevented a move line from being generated for the newly added product. As a result, the consumed quantity was 0. To get the consumed quantity, a move line is required, which is why I added this condition.

Sorry for the confusion; I have now applied it to the MRP.

Thank You.